app/code/Magento/Quote/Observer/Frontend/Quote/Address/CollectTotalsObserver.php

```diff
@@ -78,6 +89,7 @@ public function __construct(
         $this->groupManagement = $groupManagement;
         $this->addressRepository = $addressRepository;
         $this->customerSession = $customerSession;
+        $this->state = $state ?: ObjectManager::getInstance()->get(State::class);
```

Comment:
Once this PR will be delivered in a major release of Magento we can avoid using BC way of adding new dependencies to the constructor of non **@api** classes.